src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,223 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid tool */
+class GridSpaceAction : public QWidgetAction
+{
+public:
+    GridSpaceAction(QObject* parent) : QWidgetAction(parent) {
+    }
+
+protected:
+    QWidget* createWidget(QWidget* parent) override
+    {
+        auto* sizeLabel = new QLabel(QString::fromLatin1("Size"));
+        auto* gridSizeBox = new Gui::QuantitySpinBox();
+        gridSizeBox->setProperty("unit", QVariant(QString::fromLatin1("mm")));
+        gridSizeBox->setObjectName(QString::fromLatin1("gridSize"));
+        gridSizeBox->setToolTip(QApplication::translate("Sketcher_GridSize", "Distance between two subsequent grid lines"));
+        //Is General/GridSize correct? From taskSketcherGeneral it seems yes.
+        ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General/GridSize");
+        gridSizeBox->setValue(hGrp->GetFloat("GridSize", 10.0));
+        gridSizeBox->setSingleStep(1.0);
+        gridSizeBox->setMaximum(99999999.0);
+        gridSizeBox->setMinimum(0.001);
+        gridSizeBox->setDecimals(3);
+
+        QObject::connect(gridSizeBox, qOverload<double>(&Gui::QuantitySpinBox::valueChanged),
+            [=](double val) {
+            Gui::Document* doc = Gui::Application::Instance->activeDocument();
+            assert(doc);
+            auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+            assert(sketchView);
+
+            sketchView->GridSize.setValue(val);
+        });
+
+        QWidget* gridSizeW = new QWidget(parent);
+        QHBoxLayout* layout = new QHBoxLayout(gridSizeW);
+        layout->addWidget(sizeLabel);
+        layout->addWidget(gridSizeBox);
+        gridSizeW->installEventFilter(this);
+        return gridSizeW;
+    }
+    bool eventFilter(QObject* obj, QEvent* event) override

```

Comment:
Missing new line before.

Event filters should be avoided when possible. Better override the `event` function.

Reply:
This was suggested by @wwmayer so maybe he had reason to do so?

Reply:
A QAction presumably doesn't receive a suitable event when one of its widgets is about to be shown. But I haven't explicitly tested this and instead used an event filter that observes the events of the created widgets.

Reply:
Probably can use `ChildPolished` event type.
Beside, this function seems to be a lot of processing (that will be called each time any child is `Show`).
Can on explain what's the functional goal of it ? Thx

Reply:
The goal is to auto-update the value of the spin box. This value depends on which sketch is being edited. A cleaner way is to connect a function with the application's signalInEdit function and update the check boxes and spin box there.